### DISCUSSION TITLE
real time language recognition

### DISCUSSION THREAD

**Discussion by akurg** [deepgram/community#818] [General help] [answered]
How can real time Deepgram speech recognition be configured to understand any language it supports and convert to text in that language. Also, be able to switch to another language in the same session if the speaker starts using another language. We currently use Deepgram STT in real time in English successfully but it does not understand any other language we start using mid-English dialog. Thanks in advance for any help with this!

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by nikolawhallon** [comment, marked as answer]
We don't offer real-time language recognition at the moment, though we are thinking of ways to achieve this and different use-cases. For example, some clients are fine with us detecting the language once and using that language throughout the whole stream, while others are interested in code-switching where the language could change multiple times throughout a stream, and then finally there is the idea of multi-lingual models, like Spanglish or Japanglish (etc). But again, we have no concrete plans to share at this time, just noting that we are thinking about these things too.

**Reply by Nazariy995** [reply]
I think it makes a lot of sense to just detect the language on livestreaming and then use that language the whole time! Very rarely do people speak multiple languages in one sentence.

**Reply by SandraRodgers** [reply]
That's true, but this is an older post, so we've actually come out with our [Multilingual Code Switching](https://developers.deepgram.com/docs/multilingual-code-switching) feature since this was posted. It was a highly requested feature and our users have been very happy to have this as another option
